### DISCUSSION TITLE
CTk - main window fg_color "transparent" possible?

### DISCUSSION THREAD

**Discussion by Jakrich13** [TomSchimansky/CustomTkinter#1396] [Q&A] [answered]
Hey guys,
I'd like to use Ctk to create a kind of an overlay GUI, which I can lay over a map into a game. For that I need the main window to be transparent. 
I've already tried to to configure this:

app.configure(fg_color="transparent")
It raises an ValueError (f"transparency is not allowed for this attribute")

Is there any possibilty to change it in a way so I can have the main window transparent? Or any work around?

Thx in advance

**Comment by Akascape** [comment, marked as answer]
@Jakrich13 
If you want partial transparency (like an overlay) then use this method: 
```python
app.attributes('alpha', 0.5) 
```

If you want a full transparent window, then follow this (os dependent):
```python
import sys
if sys.platform.startswith("win"): # set transparency in windows
    transparent_color = '#000001' 
    app.attributes("-transparentcolor",  transparent_color) 
    app.config(bg=transparent_color)

elif sys.platform.startswith("darwin"): # set transparency in mac os
    transparent_color = 'systemTransparent' 
    app.attributes("-transparent", True) 
    app.config(bg=transparent_color)

else: 
    app.attributes('alpha', 0.5) # no full transparency method in linux
```

**Reply by Jakrich13** [reply]
Thank you for you reply, that worked perfectly for me.

**Reply by Akascape** [reply]
@Tornado300 This is only applicable for the root CTk/Tk window, not frames.
After adding this attribute to the root, you can then add the transparent color to frame parameters to make them transparent.

**Reply by RealKiryl** [reply]
@Akascape hi, how do I make separate frames transparent? Could you please provide us with an example if possible?

**Reply by Akascape** [reply]
@RealKiryl Just set fg_color to `transparent_color` of the frame.

**Reply by RealKiryl** [reply]
@Akascape thank you for your answer! And what if I'd like to make a particular frame only 20% transparent? Is there a way out for that purpose?
